rpc/ws/client.go

```diff
@@ -315,6 +379,35 @@ func (c *Client) subscribe(
 	return sub, nil
 }
 
+func (c *Client) resubscribe(requestIDs map[uint64]struct{}) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	var subscribedRequestIDs []uint64
+	for reqID := range requestIDs {
+		sub := c.subscriptionByRequestID[reqID]
+		data, err := sub.req.encode()
+		if err != nil {
+			return fmt.Errorf("subscribe: unable to encode subsciption request: %w", err)
+		}
+
+		delete(c.subscriptionByWSSubID, sub.subID)
+		sub.subID = 0 // reset subID, it will be updated later in handleNewSubscriptionMessage
+
+		c.conn.SetWriteDeadline(time.Now().Add(writeWait))
+		err = c.conn.WriteMessage(websocket.TextMessage, data)
+		if err != nil {
+			return fmt.Errorf("unable to write request: %w", err)
+		}
+		subscribedRequestIDs = append(subscribedRequestIDs, reqID)
+	}
```

Comment:
for subscribe fn, the subscriptionByRequestID is updated with each subscription as seen here https://github.com/KyberNetwork/solana-go/blob/0c161db56677e29e17fffb94e598acc4a1c7494e/rpc/ws/client.go#L369-L368. 

I don't see the same logic being apply here.

Reply:
The request ID is just a randomized number (https://github.com/KyberNetwork/solana-go/blob/0c161db56677e29e17fffb94e598acc4a1c7494e/rpc/ws/types.go#L42) to map 1-1 with sub ID during a websocket connection. I reuse request IDs so I don't have to reassign them.